monai/application/datasets.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import tarfile
+import numpy as np
+from typing import Any, Callable
+from monai.data import CacheDataset
+from monai.transforms import Randomizable
+from .utils import download_url, extractall
+
+
+class MedNISTDataset(Randomizable, CacheDataset):
+    """
+    The Dataset to automatically download MedNIST data and generate items for training, validation or test.
+    It's based on `CacheDataset` to accelerate the training process.
+
+    Args:
+        root: target dictionary to download and load MedNIST dataset.
```

Comment:
dictionary -> directory, perhaps the arg name could be `root_dir`

Reply:
updated in the latest commit.